### PR TITLE
#1160 Enable/disable summaryPanel OK button from properties controller

### LINKED ISSUE #1160: Add a new method in properties controller to enable/disable summaryPanel `Ok` button
Add new methods in properties controller to disable summary panel's "OK" button.

```
/**
* @param panelId {name: panel.id}
* @param okDisable boolean
*/
setOkButtonDisable(panelId, okDisable)

/**
* @param panelId {name: panel.id}
*/
getOkButtonDisable(panelId)
```

For example, in test harness, pass summary panel id - `{"name": "structuretableReadonlyColumnStartValue-summary-panel"}`
Screen Shot 2022-08-23 at 9 34 50 PM

When I open summary panel - 
Screen Shot 2022-08-23 at 9 35 42 PM

OK button should be disabled - 
Screen Shot 2022-08-23 at 9 35 52 PM

### PR DESCRIPTION
**Pull request by nmgokhale**
Fixes #1160 

For testing, set summaryPanel id here in test harness - 
Screen Shot 2022-08-23 at 9 34 50 PM

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/src/common-properties/properties-controller.js [hunk 1]
```diff
@@ -1630,6 +1630,22 @@ export default class PropertiesController {
 		return this.propertiesStore.getSaveButtonDisable();
 	}
 
+	/**
+	* Enable/disable OK button for given summary panel
+	* @param panelId {name: panel.id}
+	* @param okDisable boolean
+	*/
+	setOkButtonDisable(panelId, okDisable) {
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/properties-controller.js:1638]: Would this be better named `setPrimaryButtonDisabled`?

Comment (nmgokhale) [reply canvas_modules/common-canvas/src/common-properties/properties-controller.js:1638]: Should I rename it to `setWideFlyoutPrimaryButtonDisabled` ?
`setPrimaryButtonDisabled`  may be confused with an existing method `setSaveButtonDisable`.

Comment (matthoward366) [reply canvas_modules/common-canvas/src/common-properties/properties-controller.js:1638]: That's probably better.

**Review by matthoward366 — APPROVED** [review]
Looks good

**Pull request merged**